modules/data-streams/src/test/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleFixtures.java

```diff
@@ -70,7 +83,22 @@ public static DataStream createDataStream(
             builder.put(indexMetadata, false);
             backingIndices.add(indexMetadata.getIndex());
         }
-        return newInstance(dataStreamName, backingIndices, backingIndicesCount, null, false, lifecycle);
+        for (int k = 1; k <= failureIndicesCount; k++) {
+            IndexMetadata.Builder indexMetaBuilder = IndexMetadata.builder(DataStream.getDefaultFailureStoreName(dataStreamName, k, now))
+                .settings(backingIndicesSettings)
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .creationDate(now - 3000L);
+            if (k < failureIndicesCount) {
+                // add rollover info only for non-write indices
+                MaxAgeCondition rolloverCondition = new MaxAgeCondition(TimeValue.timeValueMillis(now - 2000L));
+                indexMetaBuilder.putRolloverInfo(new RolloverInfo(dataStreamName, List.of(rolloverCondition), now - 2000L));
```

Comment:
This is not actually related to/caused by your PR, but I just now realized that the rollover info is going to get overridden by failure store rollovers because we use the data stream name as the key for the rollover info map: https://github.com/elastic/elasticsearch/blob/f45fd2df5b095a4b36dff94d48b84deb0f26931c/server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java#L2007-L2007
In this specific test, it technically doesn't matter, because both `RolloverInfo`s are the same, but this is a bug and should be addressed. The bug is here: https://github.com/elastic/elasticsearch/blob/f45fd2df5b095a4b36dff94d48b84deb0f26931c/server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java#L395-L400
Where we should make a distinction between rollover for backing and failure indices. The first possible solution that comes to mind, is adding a pre- or suffix to the key we use for the rollover info map. Alternatively, we could create a new `failureStoreRolloverInfos` field on `IndexMetaData`. What do you think?

Reply:
I had a look at this together with Jimmy and he remarked that the rollover info is set on the `IndexMetaData` (i.e. not the global/generic cluster `Metadata`). So this is, fortunately, not a problem after all :)